libcodechecker/analyze/analysis_manager.py

```diff
@@ -566,6 +568,52 @@ def __create_timeout(analyzer_process):
                          (progress_checked_num.value, progress_actions.value,
                           action.analyzer_type, source_file_name))
 
+                # Remove the previously generated error file.
+                if os.path.exists(failed_dir):
+                    err_file = os.path.join(failed_dir, result_base + '.zip')
+                    if os.path.exists(err_file):
+                        os.remove(err_file)
+
+                if skip_handler:
+                    # We need to check the plist content because skipping
+                    # reports in headers can be done only this way.
+                    plist_parser.skip_report_from_plist(result_file,
+                                                        skip_handler)
+
+                try:
+                    source_files = plistlib.readPlist(result_file)['files']
+                    for sf in source_files:
+                        content_hash = None
+                        try:
+                            content_hash = util.get_file_content_hash(sf)
+
+                            hash_found = \
+                                analysis_source_file_hash.get(sf, None)
+                            if not hash_found:
+                                analysis_source_file_hash[sf] = content_hash
+                            else:
+                                LOG.debug(
+                                        "Source file was already collected")
+                                LOG.debug("Double checking hash")
+                                if hash_found != content_hash:
+                                    LOG.debug("Source content hash mismatch, "
+                                              "source file was modified "
+                                              "during the analysis.")
+
+                        except IOError as ierr:
+                            LOG.error(
+                                "Failed to calculate content hash for " + sf)
+                            LOG.error(ierr)
+
+                except Exception as ex:
+                    LOG.error(ex)
+                    LOG.error("Failed to parse result plist")
+
+                if skip_handler:
```

Comment:
Why do we need to do this twice?

Reply:
Please read the comment below. It is done to skip reports in the header files. It needs to be done twice.

Reply:
Ok, I see what you mean. There is some code duplication one of them is not necessary.